rubicon/objc/api.py

```diff
@@ -86,27 +86,14 @@ def __init__(self, method):
 
         self.selector = libobjc.method_getName(method)
         self.name = self.selector.name
-        self.pyname = self.name.replace(b':', b'_')
         self.encoding = libobjc.method_getTypeEncoding(method)
-        self.restype, *self.argtypes = ctypes_for_method_encoding(self.encoding)
-        self.imp = libobjc.method_getImplementation(method)
-        self.func = None
-
-    def get_prototype(self):
-        """Returns a ctypes CFUNCTYPE for the method."""
-        return CFUNCTYPE(self.restype, *self.argtypes)
+        self.restype, *self.imp_argtypes = ctypes_for_method_encoding(self.encoding)
+        assert self.imp_argtypes[:2] == [objc_id, SEL]
```

Comment:
assert statements are optimized to no-op if you turn on Python VM optimization, so they're generally only advisable as a "loose" check you can forget about in high-performance conditions. Should this be a hard check with a `TypeError()` on failure?

Reply:
Hm, I'm not quite sure. I'm aware that `assert` is only meant for internal invariant checks and not for input validation. The question is which category this check falls into. There is generally no reason why `imp_argtypes[:2]` would ever be anything other than `[objc_id, SEL]` unless the Objective-C compiler or runtime has seriously messed up, so I would consider this more of an invariant check than input validation. On the other hand, we're interfacing with external code, so strictly speaking this is external input and needs to be validated properly.

I'm not strongly attached to using `assert` here - I'd be okay with replacing it with a proper check.

Reply:
I tend to agree that this is more of an invariant check - I'm having difficulty thinking of a way that this would fail in practice. We can probably leave it as is.